frontend-react/docs/proposals/0005-rewrite.md

```diff
@@ -0,0 +1,167 @@
+# 0005: Rewrite of frontend-react
+
+* Status: proposed
+* Authors: @stephenkao 
+* Deciders: 
+* Proposal date: 
+* Last updated:
+
+## Problem Statement
+
+With new eyes and hands on the ReportStream website code in the last few months, some concerns have been raised about its overall maintainability, specifically:
+
+- Difficulty in building on or changing existing logic
+- Inconsistent approach to the relationship with USWDS components and styling
+- Inconsistent organization (e.g., file system patterns, component hierarchy, routing)
+- Erroneous usage and antipatterns of core technologies (e.g., inheritance working against composition, React effects)
+- Incomplete efforts to migrate to new technologies (e.g., rest-hooks and react-query)
+- Lack of separation of concerns between many functions, components, hooks
+- Opaque visibility into erring logic during bug triage
+- Unnecessary maintenance of dead code
+
+This has resulted in:
+
+- More engineering time and energy spent on navigating suboptimal practices
+- More onboarding time for new engineers
+- Bugs frequently requiring either workarounds for temporary fixes or deeper structural changes for "true" fixes
+- Less confidence in isolated changes not adversely affecting other parts of the site, potentially leading to regressions
+
+## Decision Drivers
+
+Now would be the optimal time to undertake a rewrite -- if at all -- for the following reasons:
+
+- The website will be undergoing a content refresh in the near future, which would likely benefit from a better, more cohesive and intentional foundation of code
+- The website's usage is mostly internal at the moment, so there's less risk for external users in a rewrite
+- The new maintainers of the website have frequently experienced many of the aforementioned pain points during implementation
+
+## Considered Options
+
+NOTE: Regardless of which option we choose to go forward with, this proposal and any potential work taken on will **not** involve the following items as they exist outside the intended scope and ultimately may not be in line with our ATO:
+
+- Bundler changes: CRA ejection, Webpack alternatives (Rollup), et cetera
```

Comment:
The other options listed below I understand, but why is the bundler outside of consideration?

Reply:
For documentation, the reasoning is simple scope management for this particular proposal of work. The topic of bundler is still up for discussion in general outside of this proposal I believe.